scripts/thumbnail-generator/generate_thumbnail.py

```diff
@@ -40,13 +40,21 @@ def parse_args():
                         type=str,
                         help='A comma separated list of render purposes to include in the thumbnail. Valid values are: default, render, proxy, guide.',
                         default='default')
+    parser.add_argument('--directory', 
+                        type=str,
+                        help='A directory to generate thumbnails for all .usd, .usda, .usdc, and .usdz files. When a directory is supplied, usd-file is ignored.',
+                        default='default')
+    parser.add_argument('--directory-recursive', 
+                        action='store_true',
+                        help='Will recursively search all directories underneath a given directory, requires a directory to be set.')
```

Comment:
I personally would just make a recursive flag, like `-r` `--recursive` and have it be optional in addition to `--directory`